packages/core/src/ruleset/rule/rule.ts

```diff
@@ -157,16 +167,48 @@ export class Rule implements IRule {
         throw new ReferenceError(`Alias "${alias}" does not exist`);
       }
 
-      if (alias.length + 1 === expr.length) {
-        resolvedExpr = this.owner.aliases[alias];
+      const aliasValue = this.owner.aliases[alias];
+      let actualAliasValue;
+      if (typeof aliasValue === 'string') {
+        actualAliasValue = aliasValue;
       } else {
-        resolvedExpr = this.owner.aliases[alias] + resolvedExpr.slice(alias.length + 1);
+        actualAliasValue = this.#resolveAliasForFormats(alias, aliasValue, formats);
+      }
+
+      if (actualAliasValue.length + 1 === expr.length) {
+        resolvedExpr = actualAliasValue;
+      } else {
+        resolvedExpr = actualAliasValue + resolvedExpr.slice(alias.length + 1);
       }
     }
 
     return resolvedExpr;
   }
 
+  #resolveAliasForFormats(
+    name: string,
+    { targets }: RulesetScopedAliasDefinition,
+    formats: Set<Format> | null,
+  ): string {
+    const errorMessage = `Alias "${name}" is applicable to certain formats, but the format of the linted document is not matched`;
```

Comment:
How does this manifest itself to end user, and what behavior does it cause? If there's a test around this behavior lmk and that might tell me, just couldn't find it :).

If there is a rule that uses an alias that isn't applicable to the document being linted, will this cause the lint to not go through and/or error?

Reply:
![image](https://user-images.githubusercontent.com/9273484/136156144-016a4193-d5a3-47c5-95f3-7dfd5e655bda.png)
given the following document
```yaml
swagger: 2.0
definitions:
  User:
    type: woops
```
and the following ruleset
```yaml
formats:
  - oas2
  - oas3
aliases:
  SchemaObject:
    description: Foo
    targets:
      - formats:
          - oas3.0
          - oas3.1
        given: $.components.schemas[*]
rules:
  valid-type-schema:
    given: "#SchemaObject"
    then:
      field: type
      function: enumeration
      functionOptions:
        values:
          - string
          - boolean
          - number
          - null
          - object
          - array
```

Reply:
I can't quite tell from the screenshot - did that result in the lint failing, or is it just a log to stdout?

Reply:
@marbemac it fails, I updated the screenshot.